builtin/game/fine_pointed.lua

```diff
@@ -0,0 +1,36 @@
+
+
+
+local function fine_pointed(pos, camera_pos, pos_off, look_dir)
+	local offset, nc
+	local oc = {}
+	for c,v in pairs(pos_off) do
+		if v == 0 then
+			oc[#oc + 1] = c
+		else
+			offset = v
+			nc = c
+		end
+	end
+
+	local fine_pos = {[nc] = pos[nc] + offset}
+
+	camera_pos.y = camera_pos.y + 1.625
+
+	local f = (pos[nc] + offset - camera_pos[nc]) / look_dir[nc]
+	for i = 1,#oc do
+		fine_pos[oc[i]] = camera_pos[oc[i]] + look_dir[oc[i]] * f
+	end
+
+	camera_pos.y = camera_pos.y - 1.625
+	return fine_pos
+end
+
+function minetest.fine_pointed_position(player, pt)
```

Comment:
This indirection of a function calling another function it meaningless here since fine_pointed is `local`

I would just put the entire body of `fine_pointed` into this function instead. You wouldn't even gain an indentation level!